csp_billing_adapter/adapter.py

```diff
@@ -243,11 +242,12 @@ def main() -> None:
         time.sleep(config.query_interval)  # wait 1 cycle for usage data
 
         while True:
-            now = event_loop_handler(pm.hook, config, log)
-            log.info("Processed event loop at %s", date_to_string(now))
+            start = event_loop_handler(pm.hook, config, log)
+            log.info("Processed event loop at %s", date_to_string(start))
 
-            log.debug("Sleeping for %d seconds", config.query_interval)
-            time.sleep(config.query_interval)
+            query_interval_remainder = (get_now() - start).total_seconds()
```

Comment:
And just realised this is incorrect - when I was cleaning things up I managed remove the part where I subtracted the processing time from the query interval to get the remainder.